src/components/documentManager.ts

```diff
@@ -0,0 +1,532 @@
+import { BaseItemDto } from '../api/generated/models/base-item-dto';
+
+import { JellyfinApi } from './jellyfinApi';
+
+import { parseISO8601Date } from '../helpers';
+
+// Duration between each backdrop switch in ms
+const backdropPeriodMs: number = 30000;
+
+export abstract class DocumentManager {
```

Comment:
Sorry if I am missing something, but what value does making a class with all static private/public methods/props bring vs. a typescript module which exports the public methods (but as functions)? 

When I first started reading this I assumed because it was an abstract class there would be another class which implements it. Instead its pretty much doing the job of a module but with an extra cognitive load.

Reply:
I've tried to write a long and complicated answer, but to make it easy: It's functionally the same but I find it more intuitive when it's organized into an object. For me, it decreases cognitive load rather than increase it.

In this case, before the calls into DocumentManager, you can see in the other files just by looking at the calls that the functions are inside this class. By making sure all html manipulations are in this class, you'll see by the "DocumentManager" prefix that this call does a task related to html manipulation.
I know that you can kind of mess with the imports to get something of the same effect, but I just like this style better.

Reply:
Also I should probably move the adjacent const into the class just to avoid mixing OOP and non-OOP

Reply:
But mostly this is personal preference

Reply:
yeah its functionally the same, I guess my main problem with it is that there is no concept of an abstract class in javascript. Even classes are really just sugar on top of `Function`. But that is my personal preference and there's no technical reason to choose one way over the other. I was just curious as to why you chose to use a class, Perhaps there was something I wasn't aware of that i could learn from

I do like that you've brought everything DOM related into one place, brings more structure to the project :)

Reply:
I was thinking about my decision to leave the backdropPeriodMs variable outside the class.

I would ideally make this something configurable. So that the user connecting to this chromecast can set this property to something that suits them. Or disable it completely.

We also have some user over at reddit who would prefer to switch the whole idle background image. In this case it would make sense to be able to pull this from the server.

I'll refactor this to a setter/getter style thing so it can be done easily later on :)